### PR TITLE
Handle no InferRequestedOutput case in c++ http client library

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: src/clients/c++/library/http_client.cc [hunk 1]
```diff
@@ -330,6 +330,9 @@ HttpInferRequest::PrepareRequestJson(
     }
 
     request_json->Add("outputs", std::move(outputs_json));
+  } else {
+    return Error(
+        "request should include atleast one InferRequestedOutput object");
```
Comment (deadeyegoodwin) [inline src/clients/c++/library/http_client.cc:335]: "at least"

File: src/clients/c++/library/http_client.h [hunk 2]
```diff
@@ -319,9 +319,9 @@ class InferenceServerHttpClient : public InferenceServerClient {
   /// \param result Returns the result of inference.
   /// \param options The options for inference request.
   /// \param inputs The vector of InferInput describing the model inputs.
-  /// \param outputs Optional vector of InferRequestedOutput describing how the
-  /// output must be returned. If not provided then all the outputs in the model
-  /// config will be returned as default settings.
+  /// \param outputs The vector of InferRequestedOutput describing how the
```
Comment (GuanLuo) [inline src/clients/c++/library/http_client.h:322]: Why it can't be optional?

File: src/clients/c++/library/http_client.cc (hunk 1 above)
Comment (GuanLuo) [inline src/clients/c++/library/http_client.cc:335]: "at least"

Comment (tanmayv25) [reply src/clients/c++/library/http_client.cc]: Done

File: src/clients/c++/library/http_client.h (hunk 2 above)
Comment (tanmayv25) [reply src/clients/c++/library/http_client.h:322]: If we don't specify any InferRequestedOutput, the client fails to attach the parameters to use binary extension to the output. So, by default the server will return the output as json in response. The http client library currently doesn't support output content returned as json.

**Pull request merged**